src/System.Security.Cryptography.X509Certificates/tests/ExtensionsTests.cs

```diff
@@ -298,6 +298,19 @@ public static void EnhancedKeyUsageExtension_InvalidOid(string invalidOidValue)
             Assert.ThrowsAny<CryptographicException>(() => new X509EnhancedKeyUsageExtension(oids, false));
         }
 
+        [Fact]
+        public static void EnhancedKeyUsageExtension_ImmutableOids()
+        {
+            Oid oid1 = new Oid("1.3.6.1.5.5.7.3.1");
+            OidCollection usages = new OidCollection();
+            X509EnhancedKeyUsageExtension e = new X509EnhancedKeyUsageExtension(usages, false);
+            Assert.Equal(0, e.EnhancedKeyUsages.Count);
+            usages.Add(oid1);
+            Assert.Equal(0, e.EnhancedKeyUsages.Count);
+            e.EnhancedKeyUsages.Add(oid1);
+            Assert.Equal(0, e.EnhancedKeyUsages.Count);
+        }
```

Comment:
Please add an Assert.NotSame (not reference-equal).

Either save to two local variables, or just `Assert.NotSame(e.EnhancedKeyUsages, e.EnhancedKeyUsages)`.

Reply:
OidCollection and X509Certificates are in two different assemblies in CoreFX, so the optimal thing would require public API :)